### PR TITLE
Bump versions + fix 271

### LINKED ISSUE #271: FastX Bug with Parse_hairpin
### Description of the bug

I'm running the pipeline with several fastq files that have been QC'd and trimmed already. Running into the following error message:

```
ERROR ~ Error executing process > 'NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:PARSE_MATURE'

Caused by:
  Process `NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:PARSE_MATURE` terminated with an error exit status (255)

Command executed:

  # Uncompress FASTA reference files if necessary
  FASTA="mature.fa"
  if [ ${FASTA: -3} == ".gz" ]; then
      gunzip -f $FASTA
      FASTA=${FASTA%%.gz}
  fi
  # Remove spaces from miRBase FASTA files
  # sed -i 's, ,_,g' $FASTA
  sed '#^[^>]#s#[^AUGCaugc]#N#g' $FASTA > ${FASTA}_parsed.fa
  # TODO perl -ane 's/[ybkmrsw]/N/ig;print;' ${FASTA}_parsed_tmp.fa > ${FASTA}_parsed.fa
  
  sed -i 's# .*##' ${FASTA}_parsed.fa
  seqkit grep -r --pattern ".*hsa-.*" ${FASTA}_parsed.fa > ${FASTA}_sps.fa
  seqkit seq --rna2dna ${FASTA}_sps.fa > ${FASTA}_igenome.fa
  
  cat <<-END_VERSIONS > versions.yml
  NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:PARSE_MATURE":
      seqkit: $(echo $(seqkit 2>&1) | sed 's/^.*Version: //; s/ .*$//')
  END_VERSIONS

Command exit status:
  255

Command output:
  (empty)

Command error:
  [ERRO] fastx: invalid FASTA/Q format

```

Seems like there's an error with `fastx`, which I assume is being called by `seqkit` but I'm not quite sure where to look given this. The fastq file has the following format.

```
>@A00821:1340:H2GW7DRX3:2:2101:4562:1031 1:N:0:AGATGTAC
GNCTGGCAGTAATGTAGAGCC
```

### Command used and terminal output

```console
nextflow run nf-core/smrnaseq \
    -profile test,singularity \
    --input clean_samplesheet.csv \
    --mature https://mirbase.org/download/CURRENT/mature.fa \
    --hairpin https://mirbase.org/download/CURRENT/hairpin.fa \
    --skip_fastqc \
    --skip_multiqc \
    --genome 'GRCh37' \
    --mirtrace_species 'hsa' \
    --protocol 'illumina' \
    --outdir 'Processed'
```

### Relevant files

_No response_

### System information

Nextflow version: 23.04.1
Hardware: HPC
Executor: Sun Grid Engine
Container engine: Singularity
OS: AlmaOS 8 Linux
Version of nf-core/smrnaseq: v2.2.1-gf7022ab

### PR DESCRIPTION
**Pull request by apeltzer**
This fixes the prevalent issue with miRBase that files downloaded from there using CLI tools contain HTML tags that subsequently "kill" any tool relying on them. We added SED expressions to get rid of these, clean up the FastA files and then pass them on in the pipeline. This should also fix #271 permanently, and independent (!) on which mirbase URL is used.

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/smrnaseq/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/smrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by apeltzer** [comment]
Further investigation revealed that we have a copy of mature.fa in the testdata:

```
mature           = 'https://github.com/nf-core/test-datasets/raw/smrnaseq/reference/mature.fa'
    hairpin          = 'https://github.com/nf-core/test-datasets/raw/smrnaseq/reference/hairpin.fa'
```

Thats why the tests work, but actual code doesnt :(

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,9 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [v2.2.3](https://github.com/nf-core/smrnaseq/releases/tag/2.2.3) - 2023-09-06
+
+- [[#271]](https://github.com/nf-core/smrnaseq/issues/271) - Bugfix for parsing hairpint mature files
```
Comment (christopher-mohr) [inline CHANGELOG.md:8]:
```suggestion
- [[#271]](https://github.com/nf-core/smrnaseq/issues/271) - Bugfix for parsing hairpin and mature fasta files
```

**Review by christopher-mohr — APPROVED** [review]
One suggestions, otherwise LGTM!

File: CHANGELOG.md (hunk 1 above)
Comment (apeltzer) [reply CHANGELOG.md]: image

**Pull request merged**